File: internal/backoffice/domain/entity/product_setup.go
package entity

import "time"

const (
	ProductSetupDraftStatusDraft          = "draft"
	ProductSetupDraftStatusReadyForReview = "ready_for_review"
	ProductSetupDraftStatusPublishReady   = "publish_candidate"

	ProductSetupCandidateStatusReady         = "ready"
	ProductSetupCandidateStatusPublishedMock = "published_mock"
	ProductSetupCandidateStatusArchived      = "archived"
)

type ProductSetupDraft struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Partner     string    `json:"partner"`
	BaseCost    string    `json:"baseCost"`
	RetailPrice string    `json:"retailPrice"`
	Status      string    `json:"status"`
	Notes       string    `json:"notes"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

type ProductSetupVariant struct {
	ID     string `json:"id"`
	Label  string `json:"label"`
	Color  string `json:"color"`
	Size   string `json:"size"`
	Status string `json:"status"`
}

type ProductSetupArtworkChecklist struct {
	FrontArtwork     bool `json:"frontArtwork"`
	BackArtwork      bool `json:"backArtwork"`
	MockupReady      bool `json:"mockupReady"`
	PrintSpecChecked bool `json:"printSpecChecked"`
}

type ProductSetupCandidate struct {
	ID                 string                       `json:"id"`
	DraftID            string                       `json:"draftId"`
	Title              string                       `json:"title"`
	SKU                string                       `json:"sku"`
	Partner            string                       `json:"partner"`
	BaseCost           string                       `json:"baseCost"`
	RetailPrice        string                       `json:"retailPrice"`
	EstimatedMargin    string                       `json:"estimatedMargin"`
	Status             string                       `json:"status"`
	Channel            string                       `json:"channel"`
	UpdatedAt          time.Time                    `json:"updatedAt"`
	Variants           []ProductSetupVariant        `json:"variants"`
	ArtworkChecklist   ProductSetupArtworkChecklist `json:"artworkChecklist"`
	MerchandisingNotes string                       `json:"merchandisingNotes"`
}

type ProductSetupSnapshot struct {
	Drafts     []ProductSetupDraft     `json:"drafts"`
	Candidates []ProductSetupCandidate `json:"candidates"`
}

File: internal/ui-podzone/src/services/productSetup.ts
import { postBackofficeGraphQL } from './backofficeGraphql'

export type SetupDraft = {
  id: string
  name: string
  partner: string
  baseCost: string
  retailPrice: string
  status: string
  notes: string
  createdAt?: string
  updatedAt?: string
}

export type CandidateVariant = {
  id: string
  label: string
  color: string
  size: string
  status: string
}

export type ArtworkChecklist = {
  frontArtwork: boolean
  backArtwork: boolean
  mockupReady: boolean
  printSpecChecked: boolean
}

export type CatalogCandidate = {
  id: string
  draftId: string
  title: string
  sku: string
  partner: string
  baseCost: string
  retailPrice: string
  estimatedMargin: string
  status: string
  channel: string
  updatedAt: string
  variants: CandidateVariant[]
  artworkChecklist: ArtworkChecklist
  merchandisingNotes: string
}

export type ProductSetupSnapshot = {
  drafts: SetupDraft[]
  candidates: CatalogCandidate[]
}

type ProductSetupResult<T> =
  | { success: true; data: T }
  | { success: false; message: string }

type CreateDraftPayload = {
  name: string
  partner: string
  baseCost: string
  retailPrice: string
  status: string
  notes: string
}

type PromoteCandidatePayload = {
  draftId: string
  channel: string
  variantColor: string
  variantSize: string
  artworkChecklist: ArtworkChecklist
  merchandisingNotes: string
}

export async function getProductSetupSnapshot(): Promise<
  ProductSetupResult<ProductSetupSnapshot>
> {
  const result = await postBackofficeGraphQL<{
    productSetupSnapshot: ProductSetupSnapshot
  }>(`
    query ProductSetupSnapshot {
      productSetupSnapshot {
        drafts {
          id
          name
          partner
          baseCost
          retailPrice
          status
          notes
          createdAt
          updatedAt
        }
        candidates {
          id
          draftId
          title
          sku
          partner
          baseCost
          retailPrice
          estimatedMargin
          status
          channel
          updatedAt
          merchandisingNotes
          variants {
            id
            label
            color
            size
            status
          }
          artworkChecklist {
            frontArtwork
            backArtwork
            mockupReady
            printSpecChecked
          }
        }
      }
    }
  `)
  if (!result.success) {
    return { success: false, message: result.message }
  }
  return { success: true, data: result.data.productSetupSnapshot }
}

export async function createProductSetupDraft(
  payload: CreateDraftPayload
): Promise<ProductSetupResult<SetupDraft>> {
  const result = await postBackofficeGraphQL<{
    createProductSetupDraft: SetupDraft
  }>(
    `
      mutation CreateProductSetupDraft($input: CreateProductSetupDraftInput!) {
        createProductSetupDraft(input: $input) {
          id
          name
          partner
          baseCost
          retailPrice
          status
          notes
          createdAt
          updatedAt
        }
      }
    `,
    { input: payload }
  )
  if (!result.success) {
    return { success: false, message: result.message }
  }
  return { success: true, data: result.data.createProductSetupDraft }
}

export async function promoteProductSetupCandidate(
  payload: PromoteCandidatePayload
): Promise<ProductSetupResult<CatalogCandidate>> {
  const result = await postBackofficeGraphQL<{
    promoteProductSetupCandidate: CatalogCandidate
  }>(
    `
      mutation PromoteProductSetupCandidate(
        $input: PromoteProductSetupCandidateInput!
      ) {
        promoteProductSetupCandidate(input: $input) {
          id
          draftId
          title
          sku
          partner
          baseCost
          retailPrice
          estimatedMargin
          status
          channel
          updatedAt
          merchandisingNotes
          variants {
            id
            label
            color
            size
            status
          }
          artworkChecklist {
            frontArtwork
            backArtwork
            mockupReady
            printSpecChecked
          }
        }
      }
    `,
    { input: payload }
  )
  if (!result.success) {
    return { success: false, message: result.message }
  }
  return { success: true, data: result.data.promoteProductSetupCandidate }
}

export async function updateProductSetupCandidateStatus(
  id: string,
  status: string
): Promise<ProductSetupResult<CatalogCandidate>> {
  const result = await postBackofficeGraphQL<{
    updateProductSetupCandidateStatus: CatalogCandidate
  }>(
    `
      mutation UpdateProductSetupCandidateStatus($id: ID!, $status: String!) {
        updateProductSetupCandidateStatus(id: $id, status: $status) {
          id
          draftId
          title
          sku
          partner
          baseCost
          retailPrice
          estimatedMargin
          status
          channel
          updatedAt
          merchandisingNotes
          variants {
            id
            label
            color
            size
            status
          }
          artworkChecklist {
            frontArtwork
            backArtwork
            mockupReady
            printSpecChecked
          }
        }
      }
    `,
    { id, status }
  )
  if (!result.success) {
    return { success: false, message: result.message }
  }
  return { success: true, data: result.data.updateProductSetupCandidateStatus }
}
